src/Compendium/Compendium.jsx

```diff
@@ -0,0 +1,89 @@
+import {  useState, useEffect } from "react";
+import QuoteList from "../Components/QuoteList/QuoteList";
+import { fetchQuotes, filteredQuotes, fetchSearchQuery, randomQuotes } from "../Services/Quotes";
+import './Compendium.css'
+import Header from "../Components/Header";
+import Controls from "../Components/Controls/Controls";
+
+export default function Compendium() {
+  const [loading, setLoading] = useState(true);
+  const [quotes, setQuotes] = useState([]);
+  const [characterName, setCharacterName] = useState('');
+  const [query, setQuery ] = useState('');
+  const [numOfRandom, setNum ] = useState();
```

Comment:
make sure to always initialize state with something, even if it's not the target data type (so this could be initialized as '' or 0 or even null if you wanted to)